File: external/webroot/md/codeExample/Face-API-Match/FaceMatch.cs
﻿using System;
using System.IO;
using System.Net;
using System.Text;
using System.Web;

namespace com.baidu.ai
{
	public static class Match
	{
		// 人脸比对
		public static string match()
		{
			string token = "########";
			string base643 = FileUtils.getFileBase64("/work/ai/images/face/search3.jpg");
			string base642 = FileUtils.getFileBase64("/work/ai/images/face/search2.jpg");
			string base644 = FileUtils.getFileBase64("/work/ai/images/face/search4.jpg");
			string host = "https://aip.baidubce.com/rest/2.0/faceverify/v1/match?access_token=" + token;
			Encoding encoding = Encoding.Default;
			HttpWebRequest request = (HttpWebRequest)WebRequest.Create(host);
			request.Method = "post";
			request.ContentType = "application/x-www-form-urlencoded";
			request.KeepAlive = true;
			String str = "images=" + HttpUtility.UrlEncode(base643 + "," + base642 + "," + base644);
			byte[] buffer = encoding.GetBytes(str);
			request.ContentLength = buffer.Length;
			request.GetRequestStream().Write(buffer, 0, buffer.Length);
			HttpWebResponse response = (HttpWebResponse)request.GetResponse();
			StreamReader reader = new StreamReader(response.GetResponseStream(), Encoding.Default);
			string result = reader.ReadToEnd();
			Console.WriteLine("人脸比对:");
			Console.WriteLine(result); 
			return result;
		}
			
	}
}

File: external/webroot/md/codeExample/Face-API-Match/FaceMatch.py
# encoding:utf-8
import base64
import urllib
import urllib2

from token import access_token

'''
人脸比对接口
'''

matchUrl = "https://aip.baidubce.com/rest/2.0/face/v2/match"
# 参数images：图像base64编码,多张图片半角逗号分隔
params = {
    "images": "【图片base64编码】,【图片base64编码】"}
params = urllib.urlencode(params)
access_token = access_token.AuthService()
matchUrl = matchUrl + "?access_token=" + access_token
request = urllib2.Request(url=matchUrl, data=params)
request.add_header('Content-Type', 'application/x-www-form-urlencoded')
response = urllib2.urlopen(request)
content = response.read()
if content:
    print content
